fleetshard/pkg/fleetmanager/auth.go

```diff
@@ -51,11 +63,37 @@ type Auth interface {
 // NewAuth will return Auth that can be used to add authentication of a specific AuthType to be added to HTTP requests.
 func NewAuth(t AuthType) (Auth, error) {
 	switch t {
+	case RHSSOAuthType:
+		return newRHSSOAuth(tokenPath)
 	default:
 		return newOcmAuth(os.Getenv("OCM_TOKEN"))
 	}
 }
 
+type rhSSOAuth struct {
+	tokenFilePath string
+}
+
+func newRHSSOAuth(tokenFilePath string) (*rhSSOAuth, error) {
+	if _, err := os.Stat(tokenFilePath); err != nil {
+		return nil, err
+	}
+	return &rhSSOAuth{
+		tokenFilePath: tokenFilePath,
+	}, nil
+}
+
+func (r *rhSSOAuth) AddAuth(req *http.Request) error {
+	// The file is populated by the token-refresher, which will ensure the token is not expired.
+	contents, err := os.ReadFile(r.tokenFilePath)
+	if err != nil {
+		return err
+	}
+
+	setBearer(req, string(contents))
```

Comment:
nit-nitpick: Let's call it a token.
```suggestion
	token, err := os.ReadFile(r.tokenFilePath)
	if err != nil {
		return err
	}

	setBearer(req, string(token))
```